### DISCUSSION TITLE
QueryFilter need Send + Sync

### DISCUSSION THREAD

**Discussion by alexniver** [dimforge/rapier#644] [Q&A]
my condition, main thread run graphics, other thread run rapier,   use channel send message, can't send QueryFilter

error message
```
 `(dyn for<'a> Fn(ColliderHandle, &'a rapier3d::geometry::Collider) -> bool + 'static)` cannot be shared between 
threads safely
```

**Comment by ThierryBerger** [comment]
Hi, can you provide more info/code context on what you're attempting to do ?

Have you tried wrapping your QueryFilter in a Mutex ? or reconstructing it when the message is received ?
